src/main/resources/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineView/build-pipeline-init.js

```diff
@@ -0,0 +1,47 @@
+const buildCardTemplateSource = jQuery("#build-card-template").html();
+const projectCardTemplateSource = jQuery("#project-card-template").html();
+const refreshFrequency = parseInt(document.querySelector(".pipeline-refresh-frequency").dataset.refreshFrequency, 10);
+
+var buildPipeline = new BuildPipeline(
+    buildPipelineViewProxy,
+    Handlebars.compile(buildCardTemplateSource),
+    Handlebars.compile(projectCardTemplateSource),
+    refreshFrequency
+);
+
+
+function initializeBuildCards() {
+    document.querySelectorAll('.build-data').forEach(buildElement => {
+
+        const buildId = buildElement.dataset.buildId;
+        const buildData = JSON.parse(buildElement.dataset.buildInfo);
+        const buildStatus = buildElement.dataset.buildStatus;
+        const isManualTrigger = buildElement.dataset.isManualTrigger === 'true';
+        const nextBuildNumber = parseInt(buildElement.dataset.nextBuildNumber, 10);
+        const dependencyIds = JSON.parse(buildElement.dataset.dependencyIds);
+
+        // Generate build card
+        jQuery(buildElement).append(buildPipeline.buildCardTemplate(buildData));
+
+        // add build proxy to proxies for future use-->
+        buildPipeline.buildProxies[buildId] = window[`buildProxy_${buildId}`];
+
+        if (buildStatus === 'BUILDING') {
+            buildPipeline.showProgress(buildId, dependencyIds);
+        }
+
+        jQuery("table.pipelines").on(`show-status-${buildId}`, function () {
```

Comment:
I don't quite understand what `.on` is supposed to do. I see that it was `.bind` before.

Reply:
.bind() has been deprecated and replaced with on()
https://www.w3schools.com/jquery/event_bind.asp#:~:text=The%20bind()%20method%20was%20deprecated%20in%20version%203.0.%20Use%20the%20on()%20method%20instead.

Reply:
Oh ok, so that's jquery way of adding an event listener. Couldn't tell by looking at the first argument, guess that's a custom one.